test/watershed.jl

```diff
@@ -17,6 +17,19 @@ using ImageFiltering
     @test result.segment_labels == collect(1:2)
     @test all(label->(label==result.image_indexmap[50,50]), result.image_indexmap[26:74,26:74])
 
+    mask = trues(size(img))
+    mask[60:70, 60:70] .= false
+
+    result = watershed(img, markers, compactness=10.0, mask=mask)
```

Comment:
Does `watershed(Gray.(img), markers, compactness=10.0, mask=mask)` works normally, and how about `watershed(Gray.(ing), markers)`?